crates/networking/discv5/src/lean.rs

```diff
@@ -0,0 +1,45 @@
+use std::net::{IpAddr, Ipv4Addr};
+
+use discv5::{ConfigBuilder, ListenConfig};
+
+#[derive(Debug, Clone)]
+pub struct LeanDiscoveryConfig {
+    pub discv5_config: discv5::Config,
+    pub socket_address: IpAddr,
+    pub socket_port: u16,
+    pub discovery_port: u16,
+    pub disable_discovery: bool,
```

Comment:
I don't know why there are 3 fields for discv5 here, but as I said above socket ip/port should go into the network config. So this file should be removed